### DISCUSSION TITLE
`v-if` check issues

### DISCUSSION THREAD

**Discussion by jrappen** [vuepress/core#1347] [Q&A] [answered]
With the following:

```vue
<!-- ./docs/.vuepress/components/**.vue -->

<template v-if="x">
  <img v-if="x != ''" :src="y" :alt="z" loading="lazy">
</template>
```

the conditional checks in the `v-if`s of `template` and `img` do not work as expected.

---

Any quick fix?

**Comment by Mister-Hope** [comment, marked as answer]
Who told you v-if is allowed in root `template>`

**Reply by jrappen** [reply]
There's <https://vuejs.org/guide/essentials/conditional.html#v-if-on-template>.

**Reply by jrappen** [reply]
Ok, so:

```vue
<template>
  <template v-if="...">
    <!-- ... -->
  </template>
</template>
```

fixed the first issue with `v-if`.

---

What do I do with the `img`?

**Reply by jrappen** [reply]
Fixed the second by replacing `v-if="x != ''"` with `v-if="x"`.

**Reply by Mister-Hope** [reply]
https://vuejs.org/api/sfc-spec.html#template

The root tag in sfc is marking "parts"
